### DISCUSSION TITLE
Custom aggregation functions?

### DISCUSSION THREAD

**Discussion by andreasstrom** [pyg-team/pytorch_geometric#2118] [Q&A] [answered]
Hello! I'm working on trying to implement various new aggregation functions (i.e. functions different from the 'mean', 'max', and 'add'), but I'm having trouble knowing where to start even. 
I've found plenty of documentation regarding creating custom networks, but these have all drawn from already implemented layers. Do you know of any resources dealing with the creation of custom layers or custom aggregation functions I could read up on?

**Comment by wsad1** [comment, marked as answer]
To try out different aggregation functions, you could modify the aggregate function defined in the [MessagePassing](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.message_passing.MessagePassing) class. In general [this](https://pytorch-geometric.readthedocs.io/en/latest/notes/create_gnn.html) tutorial talks about creating your own message passing class. Further check out the [GENConv](https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/nn/conv/gen_conv.html#GENConv) function, which modifies the aggregate function to implement multiple aggregation schemes.

**Reply by andreasstrom** [reply]
Thank you! I will try to do that
